src/test/java/hashtable/HashTableTest.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    @Test
+    public void valueIsUpdatedIfOldAndNewObjectsHaveTheSameKeys() {
+        HashTable table = new HashTable(10);
+        table.put("t", 15);
+        table.put("t", 20);
+        assertEquals(20, table.getValue("t"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void impossibleToRemoveElementWithNullKey() {
+        HashTable table = new HashTable(10);
+        table.remove(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void impossibleToRemoveNotExistingElement() {
+        HashTable table = new HashTable(10);
+        table.remove("test");
+    }
+
+    @Test
+    public void possibleToDeleteElementHavingPreviousNode() {
+        HashTable table = new HashTable(10);
+        table.put("t", 15);
+        table.put(6, 20);
+        table.remove(6);
+        assertEquals("(t,15)", table.printSlot(6));
+    }
+
+    @Test
+    public void possibleToDeleteElementHavingNextNode() {
+        HashTable table = new HashTable(10);
+        table.put("t", 15);
+        table.put(6, 20);
+        table.remove("t");
+        assertEquals("(6,20)", table.printSlot(6));
+    }
+
+    @Test
+    public void possibleToDeleteTheOnlyElementInSlot() {
+        HashTable table = new HashTable(10);
+        table.put("t", 15);
+        table.remove("t");
+        assertEquals("null", table.printSlot(6));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void impossibleToGetValueForNullKey() {
+        HashTable table = new HashTable(10);
+        table.getValue(null);
+    }
+
+    @Test
+    public void nullValueIsReturnedForEmptyIndex() {
+        HashTable table = new HashTable(10);
+        assertEquals(null, table.getValue(1));
+    }
+
+    @Test
+    public void valueOfTenIsReturnedForTheFirstIndex() {
+        HashTable table = new HashTable(10);
+        table.put(0, 10);
+        table.put(1, 20);
+        assertEquals(10, table.getValue(0));
+    }
+
+    @Test
+    public void valueOfFifteenIsReturnedForTheMiddleIndex() {
+        HashTable table = new HashTable(10);
+        table.put(0, 20);
+        table.put("teeeeeeeeeeee", 15);
+        table.put("teee", 20);
+        assertEquals(15, table.getValue("teeeeeeeeeeee"));
+    }
+
+    @Test
+    public void valueOfTwentyIsReturnedForTheLastIndex() {
+        HashTable table = new HashTable(10);
+        table.put(0, 15);
+        table.put("teee", 20);
+        assertEquals(20, table.getValue("teee"));
+    }
+
+    @Test
+    public void possibleToGetIndexForStringKey() {
+        HashTable table = new HashTable(10);
+        assertEquals(6, table.getIndexForKey("t"));
+    }
+
+    @Test
+    public void possibleToGetIndexForNumberKey() {
+        HashTable table = new HashTable(10);
+        assertEquals(6, table.getIndexForKey(6));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void impossibleToGetIndexForNullKey() {
```

Comment:
This should probably return null. We can treat it as if someone requested a value for a non-existing (null) key.